src/device/wg_controller.rs

```diff
@@ -429,38 +429,37 @@ async fn consolidate_wg_peers<
                 .await?;
         }
 
-        // Check if we need to update sesion keeper
-        match (
-            session_keeper,
-            features.batching.is_some(),
-            requested_peer.batching_keepalive_interval,
-        ) {
-            (Some(sk), true, Some(interval)) => {
-                if sk.get_interval(key).await != Some(interval)
-                    || requested_peer.peer.ip_addresses != actual_peer.ip_addresses
-                {
-                    let target =
-                        build_ping_endpoint(&requested_peer.peer.ip_addresses, features.ipv6);
-                    if target.0.is_some() || target.1.is_some() {
-                        sk.add_node(
-                            *key,
-                            target,
-                            Duration::from_secs(interval as u64),
-                            batcher_threshold,
-                        )
-                        .await?;
-                    } else {
-                        telio_log_warn!("Peer {:?} has no ip address", key);
+        let is_actual_peer_proxying = is_peer_proxying(actual_peer, &proxy_endpoints);
+        let is_requested_peer_proxying = is_peer_proxying(&requested_peer.peer, &proxy_endpoints);
+
+        if let Some(sk) = session_keeper {
+            if !is_actual_peer_proxying && is_requested_peer_proxying && features.batching.is_none()
+            {
+                sk.remove_node(key).await?;
+            } else if let Some(interval) = requested_peer.batching_keepalive_interval {
+                if let Some(sk) = session_keeper {
+                    if sk.get_interval(key).await != Some(interval)
+                        || (requested_peer.peer.ip_addresses != actual_peer.ip_addresses)
+                        || (is_actual_peer_proxying != is_requested_peer_proxying)
```

Comment:
Note to self: experiment with UPNP peers by not providing UPNP provider. Local testing appears to keep it oscillating

Reply:
After investigation, this is not an issue but was rather caused by a bug in batcher which will need to be fixed. So this is a symptom, not the root cause.

The bug is out of scope of this PR but it is basically that using direct connections with batching will cause immediate downgrade due to overwritten keepalive value to 0